src/PRJobProducer/IFileSystem.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO;
+using System.Threading.Tasks;
+
+namespace PRJobProducer
+{
+    public interface IFileSystem
+    {
+        Task CreateDirectoryIfNotExists(string destination);
+        Task<bool> FileExists(string location);
+        Task<Stream> ReadFile(string source);
```

Comment:
nit: I’d have OpenRead and OpenWrite that return a stream you can read from and write to

Reply:
If I did that, I think I'd have to implement my own Stream type for AzureStorageFileSystem.OpenWrite(location). The Stream could be a pretty thin wrapper around MemoryStream that simply uploads the content on dispose, but I don't think it's worthwhile for an internal abstraction.